Descent3/terrain.cpp

```diff
@@ -293,13 +294,22 @@ void GenerateLODDeltas() {
 }
 
 // Returns a unique checksum for the current terrain geometry
+//
+// The checksum (or rather, an affine modification of it known as
+// occlusion_checksum) is stored in the level's map files, so any modification
+// here ought to not alter the output value, for all imaginable inputs.
 int GetTerrainGeometryChecksum() {
-  int total = 0;
+  unsigned int total = 0;
 
   for (int i = 0; i < TERRAIN_WIDTH * TERRAIN_DEPTH; i++) {
-    total += (Terrain_seg[i].ypos + i);
+    float yi = Terrain_seg[i].ypos + i;
```

Comment:
I don't get that. Why is `int` + `uint8_t` promoted to float? Is that an obscure C++ gotcha?

Reply:
I must have mixed it up with terrain_segment::y (which is the one that is float). addressed now.